erpnext/demo/user/hr.py

```diff
@@ -120,4 +121,42 @@ def make_sales_invoice_for_timesheet(name):
 	sales_invoice.set_missing_values()
 	sales_invoice.calculate_taxes_and_totals()
 	sales_invoice.insert()
-	sales_invoice.submit()
\ No newline at end of file
+	sales_invoice.submit()
+	
+def make_leave_allocation(max_leave, employee, leave_type):
+	leave_allocation = frappe.get_doc({
+		"doctype": "Leave Allocation",
+		"employee": employee,
+		"from_date": frappe.flags.current_date,
+		"to_date": frappe.utils.add_years(frappe.flags.current_date, 1),
+		"leave_type": leave_type,
+		"new_leaves_allocated": random.randint(1, max_leave)
+	})
+	leave_allocation.insert()
+	leave_allocation.submit()
+	frappe.db.commit()			
+
+def make_leave_application():
+	employee = get_random("Employee")
+	leave_type = frappe.get_all("Leave Type", fields=['name', 'max_days_allowed'])[0]
+	if leave_type.max_days_allowed:
+		max_leave = leave_type.max_days_allowed
+	else:
+		max_leave = 10	
+	leave_type = leave_type.name
+	
+	make_leave_allocation(max_leave, employee, leave_type)
+	
```

Comment:
no need to allocate leaves every time.

Maybe allocate leaves once (in setup?)